src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs

```diff
@@ -621,6 +621,15 @@ public virtual Task WriteAsync(string? value) =>
                 t.Item1.Write(t.Item2);
             }, new TupleSlim<TextWriter, string?>(this, value), CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
 
+        /// <summary>
+        /// Writes a string to the text stream asynchronously.
+        /// </summary>
+        /// <param name="value">The string to write. If <paramref name="value"/> is <see langword="null"/>, nothing is written.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        public Task WriteAsync(string? value, CancellationToken cancellationToken) =>
+            WriteAsync(value.AsMemory(), cancellationToken);
```

Comment:
@bartonjs, was there any discussion in API review about instead adding an implicit operator from string to `ReadOnlyMemory<char>` (just as it has one to span)? Two of the three overloads being added here would then seem unnecessary.